hudi-common/src/main/java/org/apache/hudi/common/util/CommitUtils.java

```diff
@@ -127,10 +127,7 @@ private static HoodieCommitMetadata buildMetadataFromStats(List<HoodieWriteStat>
 
   public static Option<HoodieCommitMetadata> buildMetadataFromInstant(HoodieTimeline timeline, HoodieInstant instant) {
     try {
-      TimelineLayout layout = TimelineLayout.fromVersion(timeline.getTimelineLayoutVersion());
-      HoodieCommitMetadata commitMetadata = layout.getCommitMetadataSerDe().deserialize(instant,
-          timeline.getInstantDetails(instant).get(),
-          HoodieCommitMetadata.class);
+      HoodieCommitMetadata commitMetadata = timeline.loadInstantContent(instant, HoodieCommitMetadata.class);
```

Comment:
Let's add an API `readCommitMetadata(HoodieInstant instant)` to return `HoodieCommitMetadata` directly so `loadInstantContent` with clazz is not called here.

Reply:
Fixed all the relevant issues.